rs/https_outcalls/client/src/client.rs

```diff
@@ -67,11 +63,11 @@ impl CanisterHttpAdapterClientImpl {
         rt_handle: Handle,
         grpc_channel: Channel,
         query_service: QueryExecutionService,
-        inflight_requests: usize,
         metrics_registry: MetricsRegistry,
         subnet_type: SubnetType,
     ) -> Self {
-        let (tx, rx) = channel(inflight_requests);
+        #[allow(clippy::disallowed_methods)]
```

Comment:
No memory leak should be possible here as the http call are bounded on the execution side now, but is someone fills strongly about having a bounded channel, we need to make sure we set it to a value higher than `MAX_CANISTER_HTTP_REQUESTS_IN_FLIGHT`.

Reply:
I would prefer it to be bounded. Could you add a constant here that is larger than `MAX_CANISTER_HTTP_REQUESTS_IN_FLIGHT`. It would also be nice to just have a unit test with `assert!(OUTCALLS_CLIENT_MAX_IN_FLIGHT >= MAX_CANISTER_HTTP_REQUESTS_IN_FLIGHT)` such that we don't misconfigure it in the future.

Reply:
I can do this, but please note that the exec layer is configured by a config object. If somebody instantiates it with a different value, we might still end up in the case where the channel bound is lower than the configured queue size on the execution.

Reply:
PS: It seems like I can use the right config value when setting up the ic stack. So at least, it will work for any regular stack initialization.

Reply:
Makes sense.